packages/amplify-request/src/index.js

```diff
@@ -88,14 +88,15 @@ export function options(opts = {}) {
 	});
 
 	if (proxy) {
-		const { hostname: host, pathname: path, port, protocol } = new URL(proxy);
+		const { hostname: host, pathname: path, port, protocol, username, password } = new URL(proxy);
 		const agentOpts = {
 			ca:                 opts.https.certificateAuthority,
 			cert:               opts.https.certificate,
 			host,
 			key:                opts.https.key,
 			path,
 			port,
+			auth: `${username}:${password}`,
```

Comment:
Can we set this only if username, password are provided?

Reply:
Done